### baseline ZERO.v2

- Dataset has grown 20x since previous baselines were run. I must start over with the basic models to make sure nothing within the dataset was broken between then and now.
- As with previous ZERO baseline, this is simply meant to find the absolute mae floor based on pure model ignorance. *Only velocities will be looked at.*

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("baseline_data_updated.csv")
df.head(20)

,cartridge,bullet_diameter,bullet_brand,bullet_type,bullet_weight,powder_type,charge_weight,velocity_fps,barrel_length,data_source,primer,twist_rate,coal,compressed_load,jump_to_lands,amb_temp,pressure,case_capacity,suppressed
0,223_rem,0.225,barnes,hpfb,36.0,vihtavuori_n130,21.1,3402,25.0,vihtavuori,small_rifle_generic,12.0,2.181,NaN,NaN,NaN,NaN,NaN,NaN
1,223_rem,0.225,barnes,hpfb,36.0,vihtavuori_n133,22.8,3481,25.0,vihtavuori,small_rifle_generic,12.0,2.181,NaN,NaN,NaN,NaN,NaN,NaN
2,223_rem,0.225,barnes,hpfb,36.0,vihtavuori_n135,24.7,3514,25.0,vihtavuori,small_rifle_generic,12.0,2.181,NaN,NaN,NaN,NaN,NaN,NaN
3,223_rem,0.225,barnes,hpfb,36.0,vihtavuori_n140,26.2,3425,25.0,vihtavuori,small_rifle_generic,12.0,2.181,NaN,NaN,NaN,NaN,NaN,NaN
4,223_rem,0.225,barnes,hpfb,36.0,vihtavuori_n540,25.8,3301,25.0,vihtavuori,small_rifle_generic,12.0,2.181,NaN,NaN,NaN,NaN,NaN,NaN
5,223_rem,0.225,barnes,varmint_grenade,50.0,vihtavuori_n130,19.1,2877,25.0,vihtavuori,small_rifle_generic,12.0,2.201,NaN,NaN,NaN,NaN,NaN,NaN
6,223_rem,0.225,barnes,varmint_grenade,50.0,vihtavuori_n133,20.8,2959,25.0,vihtavuori,small_rifle_generic,12.0,2.201,NaN,NaN,NaN,NaN,NaN,NaN
7,223_rem,0.225,barnes,varmint_grenade,50.0,vihtavuori_n135,21.9,2976,25.0,vihtavuori,small_rifle_generic,12.0,2.201,NaN,NaN,NaN,NaN,NaN,NaN
8,223_rem,0.225,barnes,varmint_grenade,50.0,vihtavuori_n140,23.5,2963,25.0,vihtavuori,small_rifle_generic,12.0,2.201,NaN,NaN,NaN,NaN,NaN,NaN
9,223_rem,0.225,barnes,varmint_grenade,50.0,vihtavuori_n540,23.6,2920,25.0,vihtavuori,small_rifle_generic,12.0,2.201,NaN,NaN,NaN,NaN,NaN,NaN


In [3]:
target = df["velocity_fps"].astype(float)
mean_velocity = target.mean()
mean_velocity

np.float64(2678.756237453398)

In [4]:
baseline_zerov2_predictions = np.full_like(target, mean_velocity)
baseline_zerov2_mae = np.mean(np.abs(target - baseline_zerov2_predictions))
baseline_zerov2_mae

np.float64(262.73890625255723)

### Absolute ignorance mae based on new data is 262.739. This is our new floor moving forward. All future models should perform better than this 